### ISSUE TITLE
[Bug]: App  crashes when adding custom xftp-server

### ISSUE THREAD

**Issue by B-X-M** [simplex-chat/simplex-chat#2579]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Platform

iOS

### OS version

iOS 16.5

### App version

5.1.2

### Current Behavior

When adding a custom XFTP-Server and saving the new (and successfully tested) server, the app crashes with an unspecified error and the new XFTP-Server is not saved to the configuration.

### Expected Behavior

XFTP-server gets added to the list in the app

### Steps To Reproduce

Open the xftp-config in the app
Add custom server manually
Test server
Go back to XFTP list
Leave XFTP-Server list
App asks whether to save the new server to the config, click save
Dialog displays "Error" and the config is not saved

**Comment by Rematak** [comment]
I can confirm a problem in version v5.1.3(127) when adding an XFTP server on Android version 13. The test of the server ends successfully, but there is still an error message when saving.

**Comment by epoberezkin** [comment]
Can you please share the specific server address that is crashing for you, we cannot reproduce yet. Thank you!

**Comment by B-X-M** [comment]
Sorry for the late reply, but after some testing, I think I can shed some light onto this issue. It ist not related to the specific erver address.

The reported bug occurs, when the custom XFTP and SMP server are operated under the same subdomain. To reproduce:
Add a custom SMP server with the full info smp://[fingerprint]:[password]@simplex.example.com. Then add a custom XFTP server like xmp://[fingerprint]:[password]@simplex.example.com. The server verification will work, but you will not be able to save the server config. The app reports an unspecified error.

The error also occurs when adding a custom XFTP server first and then a custom SMP server. Only the first one under the same subdomain.domain.tld is saved, the second entry fails. The error does not occur however if smp and xmp entries are having their respectiv subdomains like msg.example.com and files.example.com. So it's basically not a big deal but can be quite confusing when setting up your own service.

Hope that helps. Thanks for your and the teams excellent work.

**Comment by epoberezkin** [comment]
this is helpful indeed - thank you!

**Comment by obouchet** [comment]
I can confirm that having both smp and xftp-server using the same domain is the cause of the issue. It is seen as a duplicate when saving the second server.
Having 2 different domain names pointing to the same machine and using smp server with one domain name while having xftp-server using the second domain makes the saving of servers in the app work.

**Comment by epoberezkin** [comment]
yes, or you can specify a port in one of the servers - it's the combination of hostname and port that currently has to be unique in the client.
